@potiuk because of this bug, to use the `ExternalTaskSensor` currently you must explicitly set a timeout on the sensor or your DAG will hang forever. To your point on reliance on old behavior, to workaround the bug, folks may have set that timeout to avoid an infinite hang.

In those cases, fixing this bug will cause a change in the exception they receive from `AirflowSensorTimeout` to the generic `AirflowException`. If they are relying on catching the `AirflowSensorTimeout` exception subclass they may have issues though if they catch the base class they'd still be OK.

Does that sound about right? What would you propose we do?

I'm happy to update a changelog if I'm pointed in the right direction?

Also, there are some failing checks on this PR that I don't understand. Specifically, in the Sqlite Py3.7: API Always CLI Core Integration Other Providers WWW check a test fails that I'm pretty sure I don't go anywhere near:

```
FAILED tests/jobs/test_local_task_job.py::TestLocalTaskJob::test_heartbeat_failed_fast
```

Any ideas on that front? The logs are long and I didn't see much useful in them while looking through so I wanted to ask before trying to dig deeper as I'm not super familiar with this code-base and the checks on it.

--- a/airflow/sensors/external_task.py
+++ b/airflow/sensors/external_task.py
@@ -224,7 +224,8 @@ class ExternalTaskSensor(BaseSensorOperator):
         if self.failed_states:
             count_failed = self.get_count(dttm_filter, session, self.failed_states)
 
-        if count_failed == len(dttm_filter):
+        # Fail if anything in the list has failed.
+        if count_failed > 0:
             if self.external_task_ids:
                 if self.soft_fail:
                     raise AirflowSkipException(

--- a/tests/sensors/test_external_task_sensor.py
+++ b/tests/sensors/test_external_task_sensor.py
@@ -422,12 +422,12 @@ exit 0
         # ExternalTaskSensor will also report a failure and return without
         # waiting for a timeout.
         task_chain_with_failure = ExternalTaskSensor(
-            task_id='task_chain_with_failure',
+            task_id="task_chain_with_failure",
             external_dag_id=dag_external_id,
-            external_task_id='task_external_with_failure',
+            external_task_id="task_external_with_failure",
             execution_date_fn=lambda dt: [dt + timedelta(seconds=i) for i in range(3)],
-            allowed_states=['success'],
-            failed_states=['failed'],
+            allowed_states=["success"],
+            failed_states=["failed"],
             retries=0,
             timeout=5,
             poke_interval=1,

--- a/tests/sensors/test_external_task_sensor.py
+++ b/tests/sensors/test_external_task_sensor.py
@@ -431,7 +431,7 @@ exit 0
             retries=0,
             timeout=5,
             poke_interval=1,
-            dag=dag
+            dag=dag,
         )
 
         # We need to test for an AirflowException explicitly since

